python/src/uagents/experimental/dialogues/__init__.py

```diff
@@ -50,7 +50,7 @@ def __init__(
         self.starter = False
         self.ender = False
         self._model = None
-        self._func = None
+        self._func = Optional[tuple[MessageCallback, bool]]
```

Comment:
Haven't seen this usage before, is this standard? I would've expected to see:
```python
self._func: Optional[tuple[MessageCallback, bool]] = None
```

Reply:
This is the field annotation of Pydantic V1. Fields annotated with Optional or Any were given an implicit default of None even if no default was explicitly specified. You are right in that this changed in V2 and there are no longer any type annotations that will result in a field having an implicit default value.

Meaning that I need to change it to `[...] = None`

There is a good explanation over at the Pydantic docs: https://docs.pydantic.dev/latest/migration/#required-optional-and-nullable-fields where they outline what the new syntax means but it refers to the usage of `Optional` in Pydantic models so technically there is no harm in doing what I did but I'll change it anyways to comply with the current syntax. 😃

Reply:
Oh, and yes: obviously this is a type annotation and not a declaration to "Optional" 😅